massgov/mayflower-patternlab
# DP-6522 Create "Boards and directories" patterns

## Description
Adds new components and modifies existing ones to enable generalized listings of content, and specifically allows lists of user bios. Adds support for individual user bio pages.

The new components added, General Teaser and Teaser Listing, were adapted from the Press Teaser and Press Listing respectively. Although the variables have largely been renamed to be more generalized, it longer-term it's probably desirable to deprecate Press Teaser/Listing and use General Teaser/Listing instead.

General Teaser was also adjusted to be more flexible (allowing >2 emphasized text items and allowing any components to be used in the teaser body rather than just one paragraph and one link) than is immediately required for listings of people. This should make it easier to adapt for other teasers/listings, such as search results.

## Related Issue / Ticket
https://jira.state.ma.us/browse/DP-6522

## Steps to Test
The changes have been deployed to https://tnathanjames.github.io/mayflower/
Confirm that the following Page patterns match the designs linked to in DP-6522:
1. Organization Boards (https://tnathanjames.github.io/mayflower/?p=pages-organization-boards)
2. People Listing (https://tnathanjames.github.io/mayflower/?p=pages-people-listing)
3. Bio Page (https://tnathanjames.github.io/mayflower/?p=pages-bio-page)

Some deviation from the designs (small differences in spacing, etc.) is expected when it matches established conventions, though to be safe, they should probably all be listed out to confirm that they're expected and permissible.

## Screenshots
### General Teaser molecule
<img width="814" src="https://user-images.githubusercontent.com/18170074/35174021-bf862fa4-fd3b-11e7-9800-a7aa58165c48.png">

### Teaser Listing organism
<img width="1292" src="https://user-images.githubusercontent.com/18170074/35173905-29cf7af6-fd3b-11e7-9b60-bd255d28a632.png">

### People Listing page
<img width="1327" src="https://user-images.githubusercontent.com/18170074/35173909-2d78b65e-fd3b-11e7-819c-4fada3dbb7b7.png">

### Bio Page
<img width="1266" src="https://user-images.githubusercontent.com/18170074/35173912-3113cef2-fd3b-11e7-9b19-51a82e5d7425.png">

## Additional Notes:
N/A

#### Impacted Areas in Application
* Adds additional capabilities to the Link List organism, but should be backwards-compatible without introducing any changes in behavior to current implementations.
